packages/pilot/src/pages/ErrorBoundary/reducer.js

```diff
@@ -0,0 +1,224 @@
+import {
+  __,
+  any,
+  append,
+  concat,
+  defaultTo,
+  difference,
+  either,
+  equals,
+  evolve,
+  head,
+  ifElse,
+  includes,
+  is,
+  map,
+  pipe,
+  prop,
+  props,
+  propSatisfies,
+  reject,
+  test,
+  uncurryN,
+  unless,
+  when,
+} from 'ramda'
+
+import {
+  API_ERROR_RECEIVE,
+  CLEAR_ALL_ERRORS,
+  CLEAR_ERROR,
+  CLEAR_ERRORS_BY_CODE,
+  REACT_ERROR_RECEIVE,
+} from './actions'
+import getResponseError from '../../formatters/apiError'
+import getReactError from '../../formatters/reactError'
+import localErrors from '../../errors'
+
+const initialState = []
+
+const addNewError = uncurryN(2, err => unless(
+  includes(err),
+  append(err)
+))
+
+const removeError = uncurryN(2, err => when(
+  includes(err),
+  reject(equals(err))
+))
+
+const isInCodes = codes => includes(__, codes)
+
+const hasStatusInCodes = codes => propSatisfies(isInCodes(codes), 'status')
+
+const isPrioritaryError = pipe(
+  props(['type', 'priority']),
+  equals(['apiError', true])
+)
+
+const removeErrorsByStatusCode = uncurryN(
+  2,
+  codes => reject(either(
+    hasStatusInCodes(codes),
+    isPrioritaryError
+  ))
+)
+
+const passAnyRoute = uncurryN(2, route => pipe(
+  map(item => item.test(route)),
+  any(equals(true))
+))
+
+const passExactRoute = uncurryN(2, route => pipe(
+  map(pipe(
+    regex => regex.exec(route),
+    defaultTo([]),
+    head,
+    equals(route)
+  )),
+  any(equals(true))
+))
+
+const regexTest = uncurryN(2, propName => pipe(
+  prop(propName),
+  ifElse(
+    is(RegExp),
+    test,
+    equals
+  )
+))
+
+const matchesRegex = uncurryN(2, regex => evolve({
```

Comment:
Este evolve pode ser substituido por um `where`
```suggestion
const matchesRegex = uncurryN(2, regex => where({
```
Com isso não é necessário o destructuring da linha 110 dado q o where já lhe retorna um booleano